tests/phpunit/tests/blocks/wpBlockPatternsRegistry.php

```diff
@@ -482,4 +482,80 @@ public function test_is_registered_for_known_pattern() {
 		$result = $this->registry->is_registered( 'test/one' );
 		$this->assertTrue( $result );
 	}
+
+	/**
+	 * Ensures theme patterns are registered on init.
+	 *
+	 * @ticket 59723
+	 *
+	 * @covers ::_register_theme_block_patterns
+	 */
+	public function test_register_theme_block_patterns_on_init() {
+		// This test needs to use access static class properties.
+		$registry = WP_Block_Patterns_Registry::get_instance();
+
+		// Ensure we're using a theme with patterns.
+		switch_theme( 'twentytwentythree' );
+
+		$theme          = wp_get_theme();
+		$theme_patterns = array();
+
+		foreach ( $theme->get_block_patterns() as $pattern ) {
+			$theme_patterns[] = $pattern['slug'];
+		}
+
+		// This helper is fired on the init hook.
+		_register_theme_block_patterns();
+
+		$registered = wp_list_pluck( $registry->get_all_registered(), 'name' );
+
+		// Cleanup patterns registry.
+		foreach ( $theme_patterns as $pattern ) {
+			$registry->unregister( $pattern );
+		}
+
+		foreach ( $theme_patterns as $pattern ) {
+			$this->assertContains( $pattern, $registered, 'Could not confirm theme patterns were registered' );
+		}
+	}
+
+	/**
+	 * Ensures theme patterns are not registered when not themes are active and valid.
+	 *
+	 * @ticket 59723
+	 *
+	 * @covers ::_register_theme_block_patterns
+	 */
+	public function test_register_theme_block_patterns_on_init_skipped_during_install() {
+		// This test needs to use access static class properties.
+		$registry = WP_Block_Patterns_Registry::get_instance();
+
+		// Ensure we're using a theme with patterns.
+		switch_theme( 'twentytwentythree' );
+
+		$theme          = wp_get_theme();
+		$theme_patterns = array();
+
+		foreach ( $theme->get_block_patterns() as $pattern ) {
+			$theme_patterns[] = $pattern['slug'];
+		}
+
+		/**
+		 * This will short-circuit theme activation.
+		 * @see wp_get_active_and_valid_themes().
+		 */
+		wp_installing( true );
+
+		// This helper is fired on the init hook.
+		_register_theme_block_patterns();
+
+		$registered = wp_list_pluck( $registry->get_all_registered(), 'name' );
+
+		// Cleanup.
+		wp_installing( false );
+
+		foreach ( $theme_patterns as $pattern ) {
+			$this->assertNotContains( $pattern, $registered, 'Could not confirm theme patterns were registered' );
```

Comment:
I might be wrong, but I think the message when the assertion fails should be something like `Theme patterns were incorrectly registered`.

Reply:
🤦🏻‍♂️ sloppy copy/paste when I was updating the assertions for these tests. Good catch!